File: BOJ/Dynamic_programming/BOJ_12865.py
N, K = map(int, input().split())
dp = [[0] * (K+1) for _ in range(N+1)]

for i in range(1, N+1):
    W, V = map(int, input().split())
    for j in range(K+1):
        if j - W >= 0 :
            dp[i][j] = max(dp[i-1][j], dp[i-1][j-W] + V)
        else :
            dp[i][j] = dp[i-1][j]

print(max(dp[-1]))

File: BOJ/Dynamic_programming/BOJ_12865.cpp
#include <iostream>
#include <vector>

using namespace std;

int dp[100001] = {0, };

int main(){
    int N, K;

    cin >> N >> K;

    for (int i = 0; i < N; i++){
        int W, V;
        cin >> W >> V;
        for (int i = K; i > W-1; i--){
            dp[i] = max(dp[i-W]+V, dp[i]);
        }
    }
    cout << dp[K] << endl;

    return 0;
}